parsel/selector.py

```diff
@@ -90,9 +90,15 @@ def create_root_node(
     parser_cls: Type[_ParserType],
     base_url: Optional[str] = None,
     huge_tree: bool = LXML_SUPPORTS_HUGE_TREE,
+    body: Optional[bytes] = "",
+    encoding: Optional[str] = "utf8",
 ) -> etree._Element:
     """Create root node for text using given parser class."""
-    body = text.strip().replace("\x00", "").encode("utf8") or b"<html/>"
+    if body is not None and not text:
+        body = body.replace(b"\x00", b"").strip()
+    else:
+        body = text.strip().replace("\x00", "").encode("utf8") or b"<html/>"
+
     if huge_tree and LXML_SUPPORTS_HUGE_TREE:
```

Comment:
Both here and in the `else` section the value of the encoding parameter remains hard-coded, rather than passing on the `encoding` input parameter.

Reply:
Yes. I missed part about encoding